pkg/operator/ceph/object/health.go

```diff
@@ -149,12 +152,21 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 	s3SecretKey = *user.SecretKey
 
 	if c.objectStoreSpec.IsTLSEnabled() {
-		sslSecretCert, _ := c.context.Clientset.CoreV1().Secrets(c.namespacedName.Namespace).Get(ctx, c.objectStoreSpec.Gateway.SSLCertificateRef, metav1.GetOptions{})
-		sslCert = sslSecretCert.Data[certKeyName]
+		if c.objectStoreSpec.Gateway.SSLCertificateRef != "" {
+			tlsSecretCert, err = c.context.Clientset.CoreV1().Secrets(c.namespacedName.Namespace).Get(ctx, c.objectStoreSpec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+			if tlsSecretCert != nil {
+				tlsCert = tlsSecretCert.Data[certKeyName]
+			}
+		} else if c.objectStoreSpec.GetServiceServingCert() != "" {
+			tlsCert, err = ioutil.ReadFile(ServiceServingCertCAFile)
+		}
+	}
+	if err != nil {
```

Comment:
Do not override `err`. If both fail then the second will mask the first one. Always evaluate and return errors directly.